.github/workflows/cherry-pick.yml

```diff
@@ -0,0 +1,128 @@
+            gh pr comment ${{ github.event.issue.number }} \
+              --repo ${{ github.repository }} \
+              --body 'Cherry pick action is only enabled for merged pull requests to `fasttrack/*` branches'
+          fi
+
+  # Actual cherry-pick work for each target branch
+  cherry_pick_commit:
+    needs: pr_info
+    if: |
+      needs.pr_info.outputs.status == 'MERGED' &&
+        startsWith(needs.pr_info.outputs.base_ref, 'fasttrack/')
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    defaults:
+      run:
+        shell: bash
+    name: Cherry pick
+    steps:
+      - name: Workflow trigger checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Validate and extract cherry pick target from comment
+        run: |
+          pr_comment="${{ github.event.comment.body }}"
+          branches_list=${pr_comment#'/cherry-pick '}
+          IFS=", " read -ra target_branches <<< "$branches_list"
+          echo "cherry_pick_target_branches=${target_branches[*]}" >> $GITHUB_ENV
+
+      - name: Configure local git repo
+        run: |
+          git config --local user.email "$GIT_USER_EMAIL"
+          git config --local user.name "$GIT_USER_NAME"
+        env:
+          GIT_USER_EMAIL: "cblmargh@microsoft.com"
+          GIT_USER_NAME: "CBL-Mariner Servicing Account"
+
+      # Run cherry-pick for each target branch and create new PR for it. If cherry-pick fails,
+      # output conflicts to the action logs and notify in the original PR.
+      - name: Run cherry-pick action
+        run: |
+          cp toolkit/scripts/cherry_pick.sh ${{ runner.temp }}/cherry_pick.sh
+
+          for target_branch in ${{ env.cherry_pick_target_branches }}; do
+            echo "Cherry picking commit to $target_branch"
+
+            log_file="${{ runner.temp }}/$target_branch.txt"
+
+            git reset --hard
+            git clean -df
+            git checkout -- .
+
+            ${{ runner.temp }}/cherry_pick.sh \
+              -b "$target_branch" \
+              -c "${{ needs.pr_info.outputs.merge_commit }}" \
+              -o "${{ github.event.issue.pull_request.html_url }}" \
+              -l "$log_file" \
+              -r "${{ github.repository }}" \
+              -t "${{ github.event.issue.title }}" \
+              || rc=$?
+            if [ ${rc:-0} -ne 0 ]; then
+              gh pr comment ${{ github.event.issue.number }} \
+                --repo ${{ github.repository }} \
+                --body "Cherry-pick failed for branch $target_branch. See run log for more details: $RUN_URL"
+            else
+              gh pr comment ${{ github.event.issue.number }} \
+                --repo ${{ github.repository }} \
+                --body "Cherry-pick succeeded for branch $target_branch. See pull request $(cat $log_file)"
```

Comment:
Curious: why do we print the contents of the log on a successful cherry-pick?

Reply:
I'm using the log file for multiple purposes there. If cherry-pick fails, the log file will store the merge conflicts details. Otherwise, it's used to store the output of the command `gh pr create ...` above, which is a link to the newly created PR

Reply:
Updated. The log file is no longer needed, and all `gh` commands are moved inside the bash script